systemtest/src/test/java/io/strimzi/systemtest/performance/TopicOperatorPerformanceTest.java

```diff
@@ -0,0 +1,611 @@
+                                    .addNewEnv()
+                                        .withName("STRIMZI_MAX_QUEUE_SIZE")
+                                        .withValue(String.valueOf(Integer.MAX_VALUE))
+                                    .endEnv()
+                                    .addNewEnv()
+                                        .withName("STRIMZI_MAX_BATCH_SIZE")
+                                        .withValue(maxBatchSize)
+                                    .endEnv()
+                                    .addNewEnv()
+                                        .withName("MAX_BATCH_LINGER_MS")
+                                        .withValue(maxBatchLingerMs)
+                                    .endEnv()
+                                .endTopicOperatorContainer()
+                            .endTemplate()
+                        .endEntityOperator()
+                    .endSpec()
+                    .build(),
+                    ScraperTemplates.scraperPod(testStorage.getNamespaceName(), testStorage.getScraperName()).build());
+
+            this.testStorage.addToTestStorage(TestConstants.SCRAPER_POD_KEY,
+                    kubeClient().listPodsByPrefixInName(this.testStorage.getNamespaceName(), testStorage.getScraperName()).get(0).getMetadata().getName());
+
+            // Create topics related to clients a
+            if (numberOfClientInstances != 0) {
+                KafkaTopicScalabilityUtils.createTopicsViaK8s(testStorage.getNamespaceName(), testStorage.getClusterName(), clientExchangeMessagesTopicPrefix,
+                    numberOfClientInstances, 12, 3, 1);
+
+                KafkaTopicScalabilityUtils.waitForTopicsReady(testStorage.getNamespaceName(), clientExchangeMessagesTopicPrefix, numberOfClientInstances);
+            } // create KafkaUser for TLS communication via clients
+            resourceManager.createResourceWithWait(KafkaUserTemplates.tlsUser(testStorage).build());
+
+            if (withClientsEnabled) {
+                startSendRecvTimeMs = System.currentTimeMillis();
+
+                spawnMultipleProducerAndConsumers(topicNamePrefix, numberOfClientInstances);
+            }
+
+            // -- Metrics POLL --
+            // Assuming 'testStorage' contains necessary details like namespace and scraperPodName
+            this.topicOperatorCollector = new TopicOperatorMetricsCollector.Builder()
+                    .withScraperPodName(this.testStorage.getScraperPodName())
+                    .withNamespaceName(this.testStorage.getNamespaceName())
+                    .withComponentType(ComponentType.TopicOperator)
+                    .withComponentName(this.testStorage.getClusterName())
+                    .build();
+
+            this.topicOperatorPollingThread = new TopicOperatorPollingThread(this.topicOperatorCollector, "strimzi.io/cluster=" + this.testStorage.getClusterName());
+            this.daemonThread = new Thread(this.topicOperatorPollingThread);
```

Comment:
btw having `daemonThread` and `topicOperatorPollingThread` is a bit confusing. Is it possible to have the `TopicOperatorPollingThread` as a "extendable" of the `Thread`? Or name it differently.

Reply:
Re-named it to `topicOperatorPollingRunnable` and `topicOperatorThread`